jv-hw-multithreading-2

ShemsheyMaxim opened the pull request.

ShemsheyMaxim committed:
added dependencies to pom.xml,
added class InitializeList,
added class CalculatorSumCallableImpl,
added class ExecutorServiceImpl

ShemsheyMaxim committed:
added class ForkJoinImpl,
renamed package from service on multithreading_2.

ShemsheyMaxim committed:
added tests for classes,
added class ExecutorServiceImplTest, ForkJoinImplTest, InitializeListTest.

ShemsheyMaxim committed:
added plugin for coverage test,
renamed package from multithreading_2 on multithreading

entertainment-veks commented on src/main/java/multithreading/ExecutorServiceImpl.java:
```diff
@@ -0,0 +1,38 @@
+                .collect(Collectors.toList());
+        try {
+            List<Future<Integer>> futures = executorService.invokeAll(callables);
+            executorService.shutdown();
```
I think it's not required

ShemsheyMaxim replied:
What are your mean?
As for me, it's  enought, because shutdown() is a void method.

RomanDubovskyi replied:
all good here

entertainment-veks commented on src/test/java/multithreading/ExecutorServiceImplTest.java:
```diff
@@ -0,0 +1,52 @@
+
+    @Test
+    public void checkCalculateSumOfPositiveValuesFromListOK() {
+        listNumbers = List.of(1, 2, 3, 4);
```
Let's test on list with 1m elements. We don't need test with this input

ShemsheyMaxim replied:
done

entertainment-veks commented on src/test/java/multithreading/ForkJoinImplTest.java:
```diff
@@ -0,0 +1,53 @@
+
+    @Test
+    public void checkCalculateSumOfPositiveValuesFromListOK() {
+        listNumbers = List.of(1, 2, 3, 4);
```
same

ShemsheyMaxim replied:
done

entertainment-veks commented on src/test/java/multithreading/ForkJoinImplTest.java:
```diff
@@ -0,0 +1,53 @@
+        int actual = forkJoin.compute();
+        assertNotEquals(WRONG_RESULT, actual);
+    }
+}
```
lets add newline at the end of file

ShemsheyMaxim replied:
done

entertainment-veks commented on src/test/java/multithreading/InitializeListTest.java:
```diff
@@ -0,0 +1,32 @@
+        Integer actual = list.get(0);
+        assertEquals(VALUE_LIST,actual);
+    }
+}
```
lets add newline at the end of file

ShemsheyMaxim replied:
done

ShemsheyMaxim committed:
added empty line for ForkJoinImplTest and InitializeListTest.
rewrote test methods for ExecutorServiceImplTest and ForkJoinImplTest

RomanDubovskyi commented on src/main/java/multithreading/ExecutorServiceImpl.java:
```diff
@@ -0,0 +1,38 @@
+        ExecutorService executorService = Executors.newFixedThreadPool(THREAD_POOL_SIZE);
+        List<List<Integer>> listOfSubLists = ListUtils.partition(list,
+                list.size() >= THREAD_POOL_SIZE ? list.size() / THREAD_POOL_SIZE : list.size());
+        List<CalculatorSumCallableImpl> callables = listOfSubLists.stream()
```
```suggestion
        List<Callable> tasks = listOfSubLists.stream()
```

RomanDubovskyi commented on src/test/java/multithreading/InitializeListTest.java:
```diff
@@ -0,0 +1,32 @@
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
```
it's a bad practice to use * with imports

RomanDubovskyi reviewed (approved):
you can fix comments and move to the next HW without additional review

ShemsheyMaxim committed:
fixed imports in classes InitializeListTest and ForkJoinImplTest,
changed implementation from List<CalculatorSumCallableImpl> to List<Callable<Integer>> in method execute() in class ExecutorServiceImpl
